apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -165,15 +165,28 @@ defmodule Aecore.Chain.Worker do
     Enum.each(new_block.txs, fn(tx) -> Pool.remove_transaction(tx) end)
     new_block_hash = BlockValidation.block_header_hash(new_block.header)
 
-    updated_blocks_map = Map.put(blocks_map, new_block_hash, new_block)
+    updated_blocks_map  = Map.put(blocks_map, new_block_hash, new_block)
+    hundred_blocks_map  =
+    if Kernel.map_size(updated_blocks_map) > number_of_blocks_in_memory() do
+      [{_,b} | sorted_blocks] =
+        Enum.sort(updated_blocks_map,
+          fn({_,b1}, {_,b2}) ->
+            b1.header.timestamp < b2.header.timestamp
+          end)
+      Logger.info("Block ##{b.header.height} has been removed from memory")
+      Enum.into(sorted_blocks, %{})
```

Comment:
looks like the maximum of 1 block is removed from memory, maybe it would be better to sort and then use `Enum.drop` or similar

Reply:
I'm dropping the first one from the sorted list by using pattern matching.

Reply:
okay, is good for now

Reply:
Actually I'm dropping the second element now in order to keep the genesis block as well.